internal/web/server.go

```diff
@@ -304,41 +319,37 @@ func (s *server) packages(w http.ResponseWriter, r *http.Request) {
 		fmt.Fprintf(os.Stderr, "could not load packages: %v\n", err)
 		return
 	}
-	updatesAvailable := false
+
+	// Call isUpdateAvailable for each installed package.
+	// This is not the same as getting all updates in a single transaction, because some dependency
+	// conflicts could be resolvable by installing individual packages.
+	packageUpdateAvailable := map[string]bool{}
 	for _, pkg := range packages {
-		if pkg.Package != nil && pkg.Package.Spec.PackageInfo.Version != pkg.LatestVersion {
-			updatesAvailable = true
-			break
-		}
+		packageUpdateAvailable[pkg.Name] = pkg.Package != nil && s.isUpdateAvailable(r.Context(), pkg.Name)
 	}
+
 	err = pkgsPageTmpl.Execute(w, &map[string]any{
-		"CurrentContext":   s.rawConfig.CurrentContext,
-		"Packages":         packages,
-		"UpdatesAvailable": updatesAvailable,
+		"CurrentContext":         s.rawConfig.CurrentContext,
+		"Packages":               packages,
+		"PackageUpdateAvailable": packageUpdateAvailable,
+		"UpdatesAvailable":       s.isUpdateAvailable(r.Context()),
 	})
 	checkTmplError(err, "packages")
 }
 
 func (s *server) packageDetail(w http.ResponseWriter, r *http.Request) {
 	pkgName := mux.Vars(r)["pkgName"]
-	pkg, status, manifest, latestVersion, err := describe.DescribePackage(r.Context(), pkgName)
+	pkg, status, manifest, _, err := describe.DescribePackage(r.Context(), pkgName)
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "An error occurred fetching package details of %v: \n%v\n", pkgName, err)
 		return
 	}
-	if latestVersion == "" {
-		// TODO have a look at handling latestVersion as return value from DescribePackage again – seems weird
-		latestVersion, err = repo.GetLatestVersion("", pkgName)
-	}
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "An error occurred fetching latest version of %v: \n%v\n", pkgName, err)
-	}
 	err = pkgPageTmpl.Execute(w, &map[string]any{
-		"CurrentContext": s.rawConfig.CurrentContext,
-		"Package":        pkg,
-		"Status":         status,
-		"Manifest":       manifest,
-		"LatestVersion":  latestVersion,
+		"CurrentContext":  s.rawConfig.CurrentContext,
+		"Package":         pkg,
+		"Status":          status,
+		"Manifest":        manifest,
+		"UpdateAvailable": s.isUpdateAvailable(r.Context(), pkgName),
```

Comment:
Good catch. I fixed this error and changed `DescribePackage` to return `nil` for the package and status if the package is not installed.

I also added a TODO comment to remove the `latestVersion` return parameter in `DescribePackage` since I think it's a little misplaced there.